api/index.ts

```diff
@@ -76,11 +76,12 @@ async function apiBuilder (cfg: APIConfig): Promise<FastifyTypebox> {
   await server.register(multipart)
 
   // pre-sync all sites
-  for (const siteId of await store.sites.keys()) {
+  const allSites = await store.sites.keys()
+  await Promise.all(allSites.map(async (siteId) => {
```

Comment:
maybe a todo for later to use a work queue or something? are there performance downsides to waiting on a lot of promises at once?

Reply:
Yeah, defs a TODO.

I think the main bottleneck is the IPFS add since it's a major resource hog, but I think we can get to it once we have dozens of sites and start hitting the limitations. Generally the DHT has been a major bottleneck for IPFS